core/client/fs/src/main/java/alluxio/client/block/stream/BlockOutStream.java

```diff
@@ -56,10 +54,10 @@ public static BlockOutStream createShortCircuitBlockOutStream(long blockId, long
     Closer closer = Closer.create();
     try {
       BlockWorkerClient client = closer.register(context.createBlockWorkerClient(workerNetAddress));
-      PacketOutStream outStream = PacketOutStream
-          .createLocalPacketOutStream(client, blockId, blockSize, options);
-      closer.register(outStream);
-      return new BlockOutStream(outStream, blockId, blockSize, client, options);
+      PacketOutStream outStream = closer.register(PacketOutStream
+          .createLocalPacketOutStream(context, workerNetAddress, client.getSessionId(), blockId,
+              blockSize, options));
```

Comment:
We need to have logic to generate session ID on the server side. As I mentioned to you earlier, i think it is better to do that separately.